### PR TITLE
Feat/pokemon search

### PR DESCRIPTION
**Pull request by Leeosteves**
# 🚨 Fill in the fields below

[Filter Pokemons by name](https://trello.com/c/KWrPu4Kg/48-filter-pokemons-by-name)

## Description

```
~>Added search bar for pokemons by name.
```

## Type of change

Please delete options that are not relevant.

- [x] New feature (non-breaking change which adds functionality)

- [x] PRODUCT

  - [x] Put PR link on ticket
  - [x] Move the ticket

- [x] DEVELOPERS

  - [x] My code follows the style guidelines of this project
  - [x] I have performed a self-review of my own code
  - [x] I have commented my code, particularly in hard-to-understand areas
  - [x] I have made corresponding changes to the documentation
  - [x] My changes generate no new warnings and I added `TODO:`, `FIXME:`

- [x] REVIEW
  - [x] Add reviewers: The entire team and most influential people in the area/project
  - [x] Assignees myself to it

🚨 - `Approve: Wait at least one influential in the project or more experienced`

# FRONTEND

## Screenshot or video of your application

🚨 - `Fill with your application

https://user-images.githubusercontent.com/106574167/184601597-5522ab22-3848-4917-a2c9-0eb6acefdbba.mp4

### CODE REVIEW

File: src/screens/Home/index.tsx [hunk 1]
```diff
@@ -27,18 +28,30 @@ const Home: React.FC = () => {
...
+        left={<HeaderTitle>Pokedex</HeaderTitle>}
       />
-      <Separator height={40} />
-      <HeaderTitle>Pokedex</HeaderTitle>
-      <Separator height={30} />
+      <Separator height={45} />
+      <Input
+        onChangeText={e => setTextName(e)}
+        backgroundActiveColor="#f1f1f1"
+        placeholderTextColor="black"
```
Comment (IbraDajani) [inline src/screens/Home/index.tsx:37]: placeholderColor should never be a strong color like black

it seems like the placeholder is the text itself

change it to something like a light grey

File: src/screens/Home/index.tsx [hunk 2]
```diff
@@ -27,18 +28,30 @@ const Home: React.FC = () => {
...
+        placeholder="Search your pokemon"
+        icon="search"
+        color="black"
```
Comment (IbraDajani) [inline src/screens/Home/index.tsx:40]: the color of the text and the placeholder with the same color is not a good idea

File: src/screens/Home/useHomeController.ts [hunk 3]
```diff
@@ -1,11 +1,35 @@
...
+    if (textName === '') {
+      setNewData(apiJSON);
+    } else {
+      setNewData(
+        apiJSON.filter(e => {
+          return e.name.toLowerCase().indexOf(textName.toLowerCase()) > -1;
+        }),
+      );
+    }
+  }, [textName]);
```
Comment (IbraDajani) [inline src/screens/Home/useHomeController.ts:28]: Daaaamm...

nice logic


just change the comments from Hooks to Effects

**Review by IbraDajani — CHANGES_REQUESTED** [review]

File: src/screens/Home/index.tsx (hunk 1 above)
Comment (Leeosteves) [reply src/screens/Home/index.tsx]: resolved in commit d8afbe80ecd6ce00c4f340f562705183e99e0c85

File: src/screens/Home/index.tsx (hunk 2 above)
Comment (Leeosteves) [reply src/screens/Home/index.tsx]: resolved in commit d8afbe80ecd6ce00c4f340f562705183e99e0c85

File: src/screens/Home/useHomeController.ts (hunk 3 above)
Comment (Leeosteves) [reply src/screens/Home/useHomeController.ts]: resolved in commit d8afbe80ecd6ce00c4f340f562705183e99e0c85

**Review by IbraDajani — APPROVED** [review]

**Pull request merged**